src/file.ts

```diff
@@ -2288,18 +2293,69 @@ class File extends ServiceObject<File> {
     const expiresInSeconds =
         Math.round(expiresInMSeconds / 1000);  // The API expects seconds.
 
-    const config: GetSignedUrlConfigInternal = Object.assign({}, cfg);
-
-    config.action = ({
+    const verb = ({
       read: 'GET',
       write: 'PUT',
       delete: 'DELETE',
       resumable: 'POST',
-    } as {[index: string]: string})[config.action];
+    } as {[index: string]: string})[cfg.action];
```

Comment:
this cast feels weird - is there an interface for what you're doing?

Reply:
Added an Enum type.